packages/snaps-sdk/src/types/methods/notify.ts

```diff
@@ -17,14 +32,47 @@ export enum NotificationType {
  * @property type - The type of notification to display.
  * @property message - The message to display in the notification.
  */
-export type NotifyParams = {
-  type: EnumToUnion<NotificationType>;
-  message: string;
-};
+export type NotifyParams =
+  | {
+      type: EnumToUnion<NotificationType.Native>;
+      message: string;
+    }
+  | {
+      type: EnumToUnion<NotificationType.InApp>;
+      message: string;
+    }
+  | {
+      type: EnumToUnion<NotificationType.InApp>;
+      message: string;
+      content: NotificationComponent;
+      title: string;
+    }
+  | {
+      type: EnumToUnion<NotificationType.InApp>;
+      message: string;
+      content: NotificationComponent;
+      title: string;
+      footerLink: { href: string; text: string };
+    };
 
 /**
  * The result returned by the `snap_notify` method.
  *
  * This method does not return anything.
  */
 export type NotifyResult = null;
+
+export type NotificationComponent =
+  | BoxElement
```

Comment:
Hmm, the types for this will be somewhat problematic as they won't match what is actually allowed in `Box` 🤔

Reply:
Yeah, I'm not sure exactly how to type this because I don't think I can get around typing the `Box` element with the subset of children as the same name as the regular `Box` element.